libs/core/src/components/sage-checkbox/sage-checkbox.tsx

```diff
@@ -0,0 +1,117 @@
+import { Component, h, Prop, Host, Event, EventEmitter } from '@stencil/core';
+
+@Component({
+  tag: 'sage-checkbox',
+  styleUrl: 'sage-checkbox.scss',
+  shadow: true,
+})
+export class SageCheckbox {
+  /**
+   * Whether or not the checkbox is checked.
+   */
+  @Prop() checked: boolean;
+
+  /**
+   * Whether or not the checkbox is disabled.
+   */
+  @Prop() disabled: boolean;
+
+  /**
+   * Whether or not the checkbox is invalid.
+   */
+  @Prop() invalid: boolean;
+
+  /**
+   * String used for checkbox `id` attribute and label `for` attribute.
+   */
+  @Prop() componentId: string;
+
+  /**
+   * String used for label text next to checkbox.
+   */
+  @Prop() label: string;
+
+  /**
+   * String used for helper message below checkbox.
+   */
+  @Prop() helperMessage: string;
+
+  /**
+   * String used for checkbox `name` attribute.
+   */
+  @Prop() name: string;
+
+  /**
+   * If `true`, the checkbox will visually appear as indeterminate.
+   * Only JavaScript can set the objects `indeterminate` property. See [MDN Web Docs](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input/checkbox#indeterminate_state_checkboxes).
+   */
+  @Prop() indeterminate: boolean;
+
+  /**
+   * Whether or not the checkbox is required.
+   */
+  @Prop() required: boolean;
+
+  /**
+   * The value of the checkbox that is submitted with a form.
+   */
+  @Prop() value: string;
+
+  /**
+   * Emits a boolean indicating whether the checkbox is currently checked or unchecked.
+   */
+  @Event() sageCheckboxChange: EventEmitter<boolean>;
+
+  private handleCheckboxChange(event: Event) {
+    if (this.disabled) {
+      return;
+    }
+
+    const target = event.target as HTMLInputElement;
+    const isChecked = target.checked;
+
+    this.sageCheckboxChange.emit(isChecked);
+  }
+
+  private classNames() {
+    let className = `sage-checkbox`;
+
+    if (this.invalid) {
+      const invalidClassName = 'is-invalid';
+      className += ' ' + invalidClassName;
+    }
+
+    if (this.indeterminate) {
+      const indeterminateClassName = 'is-indeterminate';
+      className += ' ' + indeterminateClassName;
+    }
+
+    if (this.disabled) {
+      const disabledClassName = 'is-disabled';
+      className += ' ' + disabledClassName;
+    }
+
+    return className;
+  }
+
+  render() {
+    return (
+      <Host>
+        <div class={this.classNames()}>
```

Comment:
Could we assign the class to `<Host class={this.classNames()}>`? If I recall, `<Host>` will output a `<div>`

Reply:
The <HOST> does not output a div and complicates applying the styles since the classes are then applied to elements outside the shadow root. Let me know if I may be misunderstanding the ask.

Reply:
When I move the classes to the host, the classes then get applied to the element outside of the shadow-root, which removed/broke all the styles nested/applied to the current class. 

I tried nesting those same styles under the :host to no avail. Maybe my set up is incorrect. I'd be happy to pair up if you have some time.